c10/xpu/XPUFunctions.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <c10/core/Device.h>
+#include <c10/xpu/XPUDeviceProp.h>
+#include <c10/xpu/XPUMacros.h>
+
+// The naming convention used here matches the naming convention of torch.xpu
+
+namespace c10::xpu {
+
+// Log a warning only once if no devices are detected.
+C10_XPU_API DeviceIndex device_count();
+
+// Throws an error if no devices are detected.
+C10_XPU_API DeviceIndex device_count_ensure_non_zero();
+
+// If this function fails, return -1. Otherwise, return the number of Intel GPUs
+// without the limitation of SYCL runtime in multi-processing.
```

Comment:
What's the meaning of "without the limitation of SYCL runtime in multi-processing."?

Reply:
refine.